purchase_sale_stock_inter_company/models/stock_picking.py

```diff
@@ -19,26 +19,61 @@ def action_done(self):
             purchase = pick.sale_id.auto_purchase_order_id
             if not purchase:
                 continue
-            purchase.picking_ids.write({"intercompany_picking_id": pick.id})
-            if not pick.intercompany_picking_id and purchase.picking_ids[0]:
-                pick.write({"intercompany_picking_id": purchase.picking_ids[0]})
+            po_picking_pending = purchase.picking_ids.filtered(
+                lambda x: x.state not in ["done", "cancel"]
+            )
+            po_picking_pending.write({"intercompany_picking_id": pick.id})
```

Comment:
Although it gets the same result, I consider dot notation cleaner:

```suggestion
            po_picking_pending.intercompany_picking_id = pick.id
```

and why writing this on each loop? It seems a bit weird this structure.

Reply:
Sorry, but I don’t understand. I only adapted the current code, but I don’t know the reasons. I think the action_done method can expect many records, and it’s better to iterate in a for loop.